evalml/automl/automl_search.py

```diff
@@ -880,7 +898,27 @@ def train_pipelines(self, pipelines):
             Note that the any pipelines that error out during training will not be included in the dictionary
             but the exception and stacktrace will be displayed in the log.
         """
-        return self._engine.train_batch(pipelines)
+        check_all_pipeline_names_unique(pipelines)
+        fitted_pipelines = {}
+        computations = []
+        for pipeline in pipelines:
+            computations.append(self._engine.submit_training_job(self.automl_data, pipeline, self.X_train, self.y_train))
+
+        while computations:
+            computation = computations.pop(0)
+            if computation.done():
+                try:
+                    fitted_pipeline = computation.get_result()
+                    fitted_pipelines[fitted_pipeline.name] = fitted_pipeline
+                except Exception as e:
+                    logger.error(f'Train error for {pipeline.name}: {str(e)}')
+                    tb = traceback.format_tb(sys.exc_info()[2])
+                    logger.error("Traceback:")
+                    logger.error("\n".join(tb))
+            else:
+                computations.append(computation)
```

Comment:
It's not exactly the same because the thing you get back from `get_result` is not the same between training and scoring and the logic for handling exceptions is not the same as well.

This might be a good spot for @kmax12 's suggestion to use `as_completed` or something like it.

If we do that, I think this will look like:

```python
    for computation in as_completed(computations):
                try:
                    fitted_pipeline = computation.get_result()
                    fitted_pipelines[fitted_pipeline.name] = fitted_pipeline
                except Exception as e:
                    logger.error(f'Train error for {pipeline.name}: {str(e)}')
                    tb = traceback.format_tb(sys.exc_info()[2])
                    logger.error("Traceback:")
                    logger.error("\n".join(tb))
```

I think at that point we're not reusing the while-loop logic so the differences between the two methods might be clearer.

Reply:
Filed https://github.com/alteryx/evalml/issues/2096 to track consolidating these two while loops!